src/EVM.hs

```diff
@@ -62,10 +62,10 @@ import Crypto.PubKey.ECC.Generate (generateQ)
 -- * Data types
 
 -- | EVM failure modes
-data Error
+data EVMError
   = BalanceTooLow W256 W256
-  | UnrecognizedOpcode Word8
-  | SelfDestruction
+  | InvalidOpcode Word8
```

Comment:
The `Invalid` opcode is (just by convention) a special opcode (`0xfe`), wheras there are many unrecognised opcodes. Prior to solc 0.8, failed `assert` statements would result in the execution of `0xfe` (which would also consume all remaining gas). These days they revert with a specific error code. 

It's not a huge deal, but I somewhwat prefer the older name, since it is more in line with this conventional usage of `Invalid` in the context of the EVM.